Add privacy implantation for logging methods

SharonHart opened the pull request:
# Pull Request Template

## What are you trying to address

- Describe the current behavior that you are modifying, and link to issue number
- If you don't have an issue, browse through existing issues to see if this is already tracked as an issue, to assign yourself to the issue and also verify that no one else is already working on the issue.

## Description of new changes

- Write a detailed description of all changes - and if appropriate, why they are needed

## For all pull requests

- [ ] Link to the issue you are solving (so it gets closed)
- [ ] Label the pull request with the appropriate area(s)
- [ ] Assign potential reviewers (you may also want to contact them on Teams to ensure timely reviews)
- [ ] Assign the project - Engineering Playbook Backlog
- [ ] Assign the pull request to yourself

TessFerrandez commented on privacy/recipes/logging.md:
````diff
@@ -33,3 +33,67 @@ class Redactor:
+    ...
+}
+```
+
+
+
+
````
```suggestion

```

TessFerrandez commented on privacy/recipes/logging.md:
````diff
@@ -33,3 +33,67 @@ class Redactor:
+```
+
+### Elastic Stack
+ Elastic Stack (formerly "ELK stack") allows logs interception by Logstash's [filter-plugins](https://www.elastic.co/guide/en/logstash/current/filter-plugins.html). Using some of the existing plugins, like 'mutate', 'alter' and 'prune' might be sufficiant for most cases of deidentifying and redacting PIIs. For a more robust and custimized usecase, a 'ruby' plugin can be used, executing arbitrary  ...
````
```suggestion

Elastic Stack (formerly "ELK stack") allows logs interception by Logstash's [filter-plugins](https://www.elastic.co/guide/en/logstash/current/filter-plugins.html). Using some of the existing plugins, like 'mutate', 'alter' and 'prune' might be sufficient for most cases of deidentifying and redacting PIIs. For a more robust and customized use-case, a 'ruby' plugin can be used, executing arbitrary Ruby code. Filter plugins also exists in some Logstash alternatives, like [Fluentd](https://docs.fluentd.org/filter) and (Fluent Bit)[https://docs.fluentbit.io/manual/pipeline/filters].
```

TessFerrandez commented on privacy/recipes/logging.md:
```diff
@@ -33,3 +33,67 @@ class Redactor:
+
+
+### Presidio
+[Presidio](https://github.com/microsoft/presidio) offers data protection and anonymization API, and can be used alogside to the log interception methods above. Presidio is containerized for REST HTTP API and also can be installed as a python package, to be called from python code. Instead of handeling the anonymization in the application code, both APIs can be used using external calls.
```
```suggestion

[Presidio](https://github.com/microsoft/presidio) offers data protection and anonymization API, and can be used alongside to the log interception methods above. Presidio is containerized for REST HTTP API and also can be installed as a python package, to be called from python code. Instead of handling the anonymization in the application code, both APIs can be used using external calls.
```

TessFerrandez commented on privacy/recipes/logging.md:
```diff
@@ -33,3 +33,67 @@ class Redactor:
+## Specific tools solutions
+
+### Application Insights
+Application Insights offers telemetry interception in some of the SDKs, that can be done by implementing the `ITelemetryProcessor` interface.
```
```suggestion

Application Insights offers telemetry interception in some of the SDKs, that can be done by implementing the `ITelemetryProcessor` interface.
```

TessFerrandez commented on privacy/recipes/logging.md:
```diff
@@ -33,3 +33,67 @@ class Redactor:
+
+### Application Insights
+Application Insights offers telemetry interception in some of the SDKs, that can be done by implementing the `ITelemetryProcessor` interface.
+ITelemetryProcessor processes the telemetry information before it is sent to Application Insights, and can be useful in many situations, such as filtering and modifiections. Below is an example of intercepting 'trace' typed telemetry:
```
```suggestion
ITelemetryProcessor processes the telemetry information before it is sent to Application Insights, and can be useful in many situations, such as filtering and modifications. Below is an example of intercepting 'trace' typed telemetry:
```

TessFerrandez reviewed (commented):
@sharonhart - is the plan to add this to the davidsb/privacy branch, and then create a PR? if so you can just merge and it will be PRd to main later. 

I added a few suggestions on some spelling, language and markdown nits if you want to incorporate them

SharonHart commented:
@TessFerrandez I understood that you are working on an initial privacy section on a different branch. If so, I'll wait for it to be merged.

TessFerrandez commented:
> @TessFerrandez I understood that you are working on an initial privacy section on a different branch. If so, I'll wait for it to be merged.

Are you talking about me personally or someone in the playbook? I am not personally adding anything around privacy

SharonHart commented:
@TessFerrandez I thought that you where working on it too, but it was by the D&AS team. 
Anyway, it might take time to be merged, so I figured it can be a good fit for the observability pillar for now. 
I've added a logging privacy appendix, included an overview and best practices, and added the tools references like in the old versioned PR. 
Would appreciate a re-review, thanks!

SharonHart commented:
@TessFerrandez After getting an update on the privacy pillar progress, I've decided to move everything to the observability pillar for the time, in a privacy section.
Since there was no context at all there, I've drafted a short description on the observability-privacy issue and practices and then state the tools recommended - presidio included.
Would be happy for another review iteration.
